### DISCUSSION TITLE
Recording computer name on which exercise was carried out?

### DISCUSSION THREAD

**Discussion by djourd1** [skranz/RTutor#37] [General]
I really appreciate the effort you put to develop RTutor. I have tested it ungraded exercises and it worked well at the University of Pretoria.

I am considering using it for graded exercises. I still have a doubt though in terms of students copying one another.

I can well imagine a situation where one bright student would do the exercises under different names, and send the submission files to some other students he wants to help. I have nothing students helping one another, but this is a situation one might want to limit.

When producing the submission file, is there a field storing the name of the computer on which the exercise was done ? If not, is there a reason (technical, ethical) on why this is not collecting such information?

Probably not essential, but just wondering ?

Damien

**Comment by skranz** [comment]
Hi Damien,

the risk of just copying another's solution without working oneself is of course an important issue one always has to think about. In my courses, I make the RTutor problem sets count between 5% and 10% of the final course grade. That seems to be enough to give incentives for most students to submit all problem sets (and from experiments much more than if the problem sets would not count towards the grade). But I also emphasize that solving the RTutor problem sets is very good training for the final exam, which accounts for the remaining 90%-95% of the grade. Indeed, in the exam there are a lot of questions (perhaps almost 50%) where students have to write or interpret R code, often also in the form that they have to fill in some placeholders. In general these exercises will be similar to my RTutor exercises. I basically tell my student that a good preparation for a large part of the exam is to be able the RTutor problem sets without hints, and try to design the exams such that this statement is indeed true.

That being said, the submission files contain fairly detailed logs of a student's solution attempts, see e.g. the documentation of RTutorSAGI here. This means in principle, you would essentially know if somebody copied the submission file by comparing these timings; it is extremely unlikely that this would be the same for two students unless they copied the submission file. If you use the RMarkdown version of the problem sets, you can, of course, not rule out that students copy the Rmd file. You could think of enforcing that the log file shows some reasonable activity, but that may be problematic if a students changes her computer and copies only the Rmd file, because then the log file may not be valid.

That being said, looking at the submissions in my courses, it seems to me that almost all students really solve the problem sets, not just copy the solution, which given the importance of the exam indeed seems the best strategy for students that want to maximize their total grade.

Sebastian
